app/services/impl/ApiTogglService.scala

```diff
@@ -2,43 +2,52 @@ package services.impl
 
 import services.TogglService
 import services.TogglService.{Project, Workspace}
+
 import play.api.libs.ws.WSClient
-import scala.concurrent.{ExecutionContext, Future}
 import play.api.libs.ws.WSAuthScheme.BASIC
 import play.api.libs.json._
 
+import scala.concurrent.{ExecutionContext, Future}
+
 /**
-  * Implimatation of [[TogglService]]
+  * Implementation of [[TogglService]]
+  *
   * @param ws the WS client
   * @param ec the execution context for asynchronous execution of program logic
   */
 class ApiTogglService(ws: WSClient)(implicit val ec: ExecutionContext) extends TogglService {
 
   /**
     * Request Toggl the list of all projects inside the specified workspace
+    *
     * @param apiToken to access a certain Toggl workspace
     * @return the list of all Toggle projects in the workspace
     */
-  override def getTogglProjects(apiToken: String): Future[List[String]] = {
-    //TODO: The workspace ID should be passed to this method.
-    // Meanwhile, in order to test change the xxxx to a roper workspace.
-
-    val request = ws.url(s"https://www.toggl.com/api/v8/workspaces/xxxx/projects")
-      .withHeaders("Content-Type" -> "application/Json")
-      .withAuth(apiToken, "api_token", BASIC)
-
-    request.get().map { response =>
-      val bodyJSValue: JsValue = Json.parse(response.body)
-      val validateBody = bodyJSValue.validate[List[Project]]
-      ws.close()
-      validateBody match {
-        case JsSuccess(projectList: List[Project], _) => projectList.map(_.name)
-        case JsError(_) => List.empty
+  override def getTogglProjects(apiToken: String): Future[List[Project]] = {
+    for {
+      workspaceID <- getTogglWorkspace(apiToken)
+      request <- ws.url(s"https://www.toggl.com/api/v8/workspaces/" + workspaceID.head + "/projects")
+        .withHeaders("Content-Type" -> "application/Json")
+        .withAuth(apiToken, "api_token", BASIC).get().map { response =>
+        val bodyJSValue: JsValue = Json.parse(response.body)
+        val validateBody = bodyJSValue.validate[List[Project]]
+        ws.close()
+        validateBody match {
+          case JsSuccess(projectList: List[Project], _) => projectList
+          case JsError(_) => List.empty
+        }
       }
-    }
+
+    } yield request
   }
 
-  override def getTogglWorkspace(apiToken: String): Future[List[String]] = {
+  /**
+    * Gets the workspace ID
+    *
+    * @param apiToken API token provided by the user
+    * @return List that always contains a single element: the workspace ID
```

Comment:
Yes, because you can have more than one workspace. But the concept of 'Toggl workspace' is not very clear for us, since we always have one, that's why we return a List of Longs, but after this method as been invoked we always make a .head to the List and use the first element. Maybe in the future we should consider the possibility of somebody having more than 1 workspace in Toggl.